### DISCUSSION TITLE
Parsing timeline elements properly

### DISCUSSION THREAD

**Discussion by collinabidi** [horovod/horovod#3101] [Q&A]
I'm parsing the output of the Horovod timeline tool (timeline.json) using a Python script rather than the chrome tracing tool. 

![timeline](https://user-images.githubusercontent.com/29078081/121730445-b046ed00-cabd-11eb-99b3-b1f0daebc931.jpg)

To illustrate my goal, I'm simply attempting to automatically find the "duration" parameter for each MPI_ALLREDUCE event in the above picture. Chrome tracing automatically generates this value, which is not present in the .json file itself. There's no documentation that I've been able to find on this. Do you know how to properly parse this?

**Comment by gkanwar** [comment]
Just stumbling on this while looking for other stuff, but I happen to have literally just looked at the relevant info!

The format is the general Trace Event format specified here: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview. As stated in that doc "duration events" consist of a beginning event ("ph": "B") followed by the next end event ("ph": "E") for a given pid/tid. So if you just pair these up you can get the duration. The rest of the information about the format is also carefully detailed in that doc.
